### PR TITLE
Issue #241. Allow users to copy a single file into a deployed context.

### PR DESCRIPTION
**Pull request by sandrapascual**
Issue #241. Allow users to copy a single file into a deployed context.
Usefull to hot change report, jsp, xml and properties files.

### CODE REVIEW

File: core/src/main/java/psiprobe/controllers/deploy/CopySingleFileController.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+	  List<Context> apps;
+	    try {
+	      apps = getContainerWrapper().getTomcatContainer().findContexts();
+	    } catch (NullPointerException ex) {
+	      throw new IllegalStateException("No container found for your server: "
+	          + getServletContext().getServerInfo(), ex);
+	    }
+	  
+	    Vector applications = new Vector();
```
Comment (hazendaz) [inline core/src/main/java/psiprobe/controllers/deploy/CopySingleFileController.java]: Can you use something other than vector here?  Vector is a bad idea for new code.

File: core/src/main/java/psiprobe/controllers/deploy/CopySingleFileController.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	      if (appContext.getName() != null && appContext.getName().trim().length()>0) {
+	    	  Map<String, String> app = new HashMap<String, String>();
+	    	  app.put("value", appContext.getName());
+	    	  app.put("label", appContext.getName());
+	    	  applications.add(app);
+	      }
+	    }
+	    request.setAttribute("apps", applications);
+
+	  if (FileUploadBase.isMultipartContent(new ServletRequestContext(request))) {
```
Comment (hazendaz) [inline core/src/main/java/psiprobe/controllers/deploy/CopySingleFileController.java]: Please format the this class.

**Comment by hazendaz** [comment]
@sandrapascual Just a couple of comments.  Otherwise looks promising :)  

Let me toss another idea out here.  This currently just addresses a deployed context, given we have pretty much full control of tomcat, and I assume this doesn't allow this, would it be possible to maybe make this even more flexible so that we could change tomcat files in bin/conf directories?  I see a need for that where password might exist we need to change rather than going to the box.  I know in my use case anyways we have the ability externally to start/stop tomcats but not to address box level changes like that.  Would be cool to have that, maybe in another pull request.  Not sure it's completely feasible but would be a nice feature.

**Comment by hazendaz** [comment]
@sandrapascual When you address the small issues I noted, please rebase against lastest master.  We have a check that prevents merging otherwise.  Just be sure to rebase right before you are ready to push this again.  I may or may not have a few other changes today :)

**Comment by sandrapascual** [comment]
Hi, I've change the Vector use and formated the class. 
Sorry, but I usually have problems to understand the rebase methods in git. I've tried to do it as better as I can. Please let me know if my pull-request is not good... 
Moving to the possibility of copying files to bin/conf folders, I think it could be dangerous for not experimented users... I could be a good option to configure correctly the permissions

**Comment by hazendaz** [comment]
No worries on rebasing.  For reference, it's quite easy...'git pull --rebase upstream master'.  Then once that completelys perform a squash interactively ('git rebase -i HEAD~X where x is number of commits to squash).  Then simply follow directions provided by git.

Since github otherwise allows squashing when pulling this in, I'll do that.  Thanks for the contribution!

**Pull request merged**